### PR TITLE
[closes #2] Removed reduce functions, Added branch coverage

### LINKED ISSUE #2: Support Branches
Currently only function counts and line counts are supported.  This should support branch data as well.

### PR DESCRIPTION
**Pull request by wmbest2**
@vokal/web please feel free to leave until tomorrow.  I just wanted to get this in so next week I can focus on working towards an upload script for vokal's coverage viewer.

### CODE REVIEW

File: source/index.js [hunk 1]
```diff
@@ -52,19 +63,35 @@ var unpackage = function ( report )
...
+                            for (var i = 0; i < count; ++i) {
+                                branches = branches.concat( {
+                                    line: Number( l.$.nr ),
+                                    block: 0,
+                                    branch: Number( i ),
+                                    taken:  i < Number( l.$.cb ) ? 1 : 0
+                                } )
+                            }
+
+                            return branches;
```
Comment (wmbest2) [inline source/index.js:90]: This was the trickiest bit.  lcov breaks it's branch data out into each traversable branch.  Jacoco only outputs information for the line.  So I need to map the lines to a "faked" array of branch data and then merge them into a single array.

For instance:

``` xml
<line nr="13" mb="1" cb="1" />
```

Would become:

``` json
[{"line": 13, "block": 0, "branch": 0, "taken": 1}, 
 {"line": 13, "block": 0, "branch": 1, "taken": 0}]
```

File: source/index.js [hunk 2]
```diff
@@ -52,19 +63,35 @@ var unpackage = function ( report )
...
+                    found: Number(branches.$.covered) + Number(branches.$.missed),
+                    hit:  Number(branches.$.covered),
+                    details: !s.line ? [] : [].concat.apply( [], 
+                        s.line.filter( function ( l ) 
+                        {
+                            return Number( l.$.mb ) > 0 || Number( l.$.cb ) > 0;
+                        })
+                        .map( function ( l )
+                        {
+                            var branches = []
```
Comment (jrit) [inline source/index.js]: missing a `;`

**Comment by jrit** [comment]
LGTM when semicolon added

**Comment by jrit** [comment]
Some of the spacing is a little off of our coding standards too

**Comment by wmbest2** [comment]
Changes made @jrit

**Pull request merged**